Created new Fortanix KMIP topic for TDE

Andriciuc opened the pull request:
Since we can use Fortanix, I added it to the config KMIP section, it tells the user where to look for more details on setting it up and provides a bit of examples with pg_tde_add_global_key_provider_kmip

Andriciuc committed:
updated with new fortanix file

Since we can use Fortanix, I added it to the config KMIP section, it tells the user where to look for more details on setting it up and provides a bit of examples with pg_tde_add_global_key_provider_kmip

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-fortanix.md:
```diff
@@ -0,0 +1,25 @@
+# Fortanix KMIP Server Configuration
+
+`pg_tde` is compatible with Fortanix Data Security Manager (DSM) via the KMIP protocol. For a full setup guide, see [the Fortanix documentation here](https://support.fortanix.com/docs/using-fortanix-data-security-manager-with-edb-postgres-for-tde).
```
that link is for the EDB Postgres fork, not ours. You probably want to link the generic KMIP key creation/setup documentation page.

Andriciuc replied:
Fixed! Updated the link to the generic KMIP client link for Fortanix: https://support.fortanix.com/docs/users-guide-account-client-configurations?highlight=KMIP#23-kmip-clients

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-fortanix.md:
```diff
@@ -0,0 +1,25 @@
+
+## Recommended Configuration Steps
+
+1. Enable KMIP for your Fortanix DSM instance.
```
If I remember the setup process correctly, it is enabled by default. All you have to do is upload keys and setup permissions, and a few other steps (follow the fortanix documentation, it's actually pretty good)

Andriciuc replied:
I updated the steps, made them more simplified

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-fortanix.md:
````diff
@@ -0,0 +1,25 @@
+```sql
+SELECT pg_tde_add_global_key_provider_kmip(
+    'fortanix_kms_provider',
+    'kms.fortanix.com',
````
the real domain won't be this, they have region specific domains for eu/us/etc.

Andriciuc replied:
To make sure the user knows that this is the case, I added a second note:

Replace ‘your-region.kms.fortanix.com’ with the actual KMIP endpoint for your Fortanix DSM instance. Fortanix uses region-specific or tenant-specific KMIP domains (e.g., eu.kms.fortanix.com, us.kms.fortanix.com). Refer to your DSM dashboard or administrator to confirm the correct value.

I also updated the example with the following change on the above line 16:
...
'your-region.kms.fortanix.com', -- Replace with your actual Fortanix DSM endpoint
...

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Update kmip-fortanix.md

updated the SELECT example with clarification and updated the link for the KMIP (generic client install)

Added a note to reinforce the region requirement

Andriciuc committed:
Merge branch 'docs-add-fortanix' of https://github.com/percona/postgres into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Update kmip-fortanix.md

updated setup step 1 from recommended steps

Andriciuc committed:
improved config steps

improved steps, still would like each step to be more descriptive though

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-fortanix.md:
```diff
@@ -0,0 +1,31 @@
+1. Obtain a client certificate, private key, and CA certificate from Fortanix DSM.
+2. Ensure you store the certificates securely.
+3. Register Fortanix DSM with `pg_tde`, see [the following example](#example-configuration-sql).
+4. Enable `pg_tde` and restart PostgreSQL, see [the following topic](../setup.md#configure-pg_tde).
```
Wrong order, first you have to create the pg tde extension/restart, andd you can configure kmip after that

Andriciuc replied:
I actually removed these steps, I understood that we are duplicating info we already have in another part of our docs. Thank you @dutow !!

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-fortanix.md:
```diff
@@ -0,0 +1,31 @@
+## Recommended Configuration Steps
+
+1. Obtain a client certificate, private key, and CA certificate from Fortanix DSM.
+2. Ensure you store the certificates securely.
```
Do we really need to mention this separately? But if we do, it should be in all doc pages, not just specific for fortanix...

Andriciuc replied:
The idea was that maybe some of the configs required specific steps to work, unique to these KMIPs, as it turns out these are not required, so I will just remove these recommended steps.

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Update kmip-fortanix.md

removed recommended config steps and I pointed the user to the Adding or modifying KMIP providers topic for examples of config files

Andriciuc committed:
Merge branch 'docs-add-fortanix' of https://github.com/percona/postgres into docs-add-fortanix

Andriciuc committed:
Update kmip-fortanix.md

removed example config title

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-fortanix.md:
```diff
@@ -0,0 +1,13 @@
+For more information on adding or modifying the provider, see the [Adding or modifying KMIP providers](https://docs.percona.com/pg-tde/functions.html?h=pg_tde_add_global_key_provider_kmip#adding-or-modifying-kmip-providers) topic.
+
+!!! note
+    Replace 'your-region.kms.fortanix.com' with the actual KMIP endpoint for your Fortanix DSM instance.
```
Now this note is out of place, we no longer mention fortanix URLs other than here. Same for the next note, we also do not mention certificate locations here.

Andriciuc replied:
Removed!

Andriciuc committed:
Update kmip-fortanix.md

removed out of date notes, not needed as we don't explain config

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-fortanix
